ee/desktop/runner/runner_darwin.go

```diff
@@ -0,0 +1,22 @@
+//go:build darwin
+// +build darwin
+
+package runner
+
+import (
+	"os/exec"
+)
+
+// For notifications to work, we must run in the user context with launchctl asuser.
+func runAsUser(uid string, cmd *exec.Cmd) error {
+
+	// the remaining code in this function is not covered by unit test since it requires root privileges
```

Comment:
If called with your own UID, I think this will work. But changing UIDs requires root.

Reply:
Added back the check from runner_linux to handle the current user not being root